Simplified/NYPLUserProfileDocument.swift

```diff
@@ -67,6 +81,32 @@ import Foundation
     dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss'Z'"
     jsonDecoder.dateDecodingStrategy = .formatted(dateFormatter)
     
-    return try jsonDecoder.decode(UserProfileDocument.self, from: data)
+    do {
+      return try jsonDecoder.decode(UserProfileDocument.self, from: data)
+    } catch let DecodingError.dataCorrupted(context) {
+      throw NSError(domain: NSCocoaErrorDomain,
```

Comment:
shouldn't we use the domain and code of the caught DecodingError instead of changing it ? that way you don't even need to define new constants.